fetcher/fedora.go

```diff
@@ -0,0 +1,293 @@
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func fetchUpdateInfosFedora(results []FetchResult) ([]FetchResult, error) {
+	log15.Info("start fetch updateinfo in repomd.xml")
+	var updateInfoReqs []fetchRequest
+	for _, r := range results {
+		var repoMd RepoMd
+		if err := xml.NewDecoder(bytes.NewBuffer(r.Body)).Decode(&repoMd); err != nil {
+			log15.Warn(fmt.Sprintf("Failed to decode repomd. Skip to fetch version %s", r.Target), "err", err)
+			continue
+		}
+
+		for _, repo := range repoMd.RepoList {
+			if repo.Type == "updateinfo" {
+				u, err := url.Parse(r.URL)
+				if err != nil {
+					return nil, err
+				}
+				u.Path = strings.Replace(u.Path, "repodata/repomd.xml", repo.Location.Href, 1)
+				req := fetchRequest{
+					url:          u.String(),
+					target:       r.Target,
+					mimeType:     mimeTypeXz,
+					concurrently: false,
+				}
+				updateInfoReqs = append(updateInfoReqs, req)
+				break
+			}
+		}
+	}
+
+	if len(updateInfoReqs) == 0 {
+		return nil, fmt.Errorf("No updateinfo field in the repomd")
+	}
+
+	results, err := fetchFeedFiles(updateInfoReqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func parseFetchResultsFedora(results []FetchResult) (fedoraUpdatesPerVersion, error) {
+	updateInfos := make(fedoraUpdatesPerVersion, len(results))
+	for _, r := range results {
+		var updateInfo FedoraUpdates
+		if err := xml.NewDecoder(bytes.NewReader(r.Body)).Decode(&updateInfo); err != nil {
+			return nil, err
+		}
+		var securityUpdate []FedoraUpdateInfo
+		for _, update := range updateInfo.UpdateList {
+			if update.Type == "security" {
+				cveIDs := []string{}
+				for _, ref := range update.References {
+					id := util.CveIDPattern.FindAllString(ref.Title, -1)
+					if id != nil {
+						cveIDs = append(cveIDs, id...)
+					}
+				}
+				if len(cveIDs) != 0 {
+					cveIDs = util.UniqueStrings(cveIDs)
+					update.CVEIDs = cveIDs
+					securityUpdate = append(securityUpdate, update)
+				}
+			}
+		}
+		updateInfo.UpdateList = securityUpdate
+		updateInfos[r.Target] = &updateInfo
+	}
+	return updateInfos, nil
+}
+
+func fetchModuleFeedFilesFedora(reqs []fetchRequest) ([]FetchResult, error) {
+	if len(reqs) == 0 {
+		return nil, fmt.Errorf("There are no versions to fetch")
+	}
+	results, err := fetchFeedFiles(reqs)
+	if err != nil {
+		return nil, xerrors.Errorf("Failed to fetch. err: %w", err)
+	}
+	return results, nil
+}
+
+func fetchModulesYamlFedora(results []FetchResult) (fedoraModuleInfosPerVersion, error) {
+	log15.Info("start fetch modules.yaml in repomd.xml")
+	var updateInfoReqs []fetchRequest
+	for _, r := range results {
+		var repoMd RepoMd
+		if err := xml.NewDecoder(bytes.NewBuffer(r.Body)).Decode(&repoMd); err != nil {
+			log15.Warn(fmt.Sprintf("Failed to decode repomd. Skip to fetch version %s", r.Target), "err", err)
+			continue
+		}
```

Comment:
That's what Amazon does, but can we go through the case where the decode of repomd.xml fails?

Reply:
I thought that I want to continue to fetch other version of fedora if the process failed to fetch some versions.
for example, if it failed to fetch fedora34, it continues to fetch fedora34 when command is `fetch fedora 34 35`

In the case to continue or quit if failed once, I think both work fine.